python/pyarrow/scalar.pxi

```diff
@@ -847,6 +882,12 @@ cdef class BinaryScalar(Scalar):
         buffer = self.as_buffer()
         return None if buffer is None else buffer.to_pybytes()
 
+    def __bytes__(self):
+        return (self.as_py())
+
+    def __buffer__(self):
+        return(memoryview(self.as_buffer()))
```

Comment:
This here will be a bit more tricky. We need to go to a lower level and pass the pointer to the actual buffer in memory, as well as some metadata. Also the method will need to be `__getbuffer__ `. I think we should be able to mostly reuse what is defined here: https://github.com/apache/arrow/blob/60b5ab9ee0bf070f03cf5c92fe0add0257543dfd/python/pyarrow/io.pxi#L1561-L1584 with the buffer being `self.as_buffer()`.